Cargo.toml

```diff
@@ -6,7 +6,7 @@ description = "Bindings for DataFusion used by Dask-SQL"
 readme = "README.md"
 license = "Apache-2.0"
 edition = "2021"
-rust-version = "1.72"
+rust-version = "1.73"
```

Comment:
All for bumping! Just for my own knowledge was this a requirement or just doing it because it has been awhile? Either way is fine and this blocks nothing just want to know.

Reply:
I actually just did this to try and triage some issues we've been seeing in the aarch64 builds around hash mismatches?

https://github.com/dask-contrib/dask-sql/actions/runs/8437291358/job/23106757782

The errors in that run seemed to imply that there was something wrong with the rust 1.72 packages, but actually it turns out that this seems to be an issue with newer versions of conda-build (and/or boa), looks like https://github.com/dask-contrib/dask-sql/pull/1319/commits/9f3248263e44c599bb8d716f2bbd4df1140effdf unblocked the builds